drivers/adc/ad7799/ad7799.c

```diff
@@ -235,17 +235,18 @@ int32_t ad7799_get_channel(struct ad7799_dev *device, uint8_t ch,
 }
 
 /**
- * @brief Read data in mV from specific ADC channel.
+ * @brief Read scaled data from specific ADC channel.
  * @param device - The device structure.
  * @param ch - The ADC channel.
- * @param data_mv - The content of the data converted in mV.
+ * @param data_scaled - The content of the scaled data
  * @return SUCCESS in case of success, negative error code otherwise.
  */
-int32_t ad7799_read_channel_mv(struct ad7799_dev *device, uint8_t ch,
-			       int32_t *data_mv)
+int32_t ad7799_read_channel_scaled(struct ad7799_dev *device, uint8_t ch,
+				   int32_t *data_scaled)
 {
 	int32_t ret;
 	uint32_t data, temp;
+	float vref_scaled = device->vref * device->scale;
```

Comment:
I'am not sure if we should use float. Some MCU can't do floating point operations